bc_obps/registration/tests/endpoints/test_operations_endpoints.py

```diff
@@ -1215,3 +1217,50 @@ def test_unapproved_industry_users_cannot_create_or_update_operations(self):
             + "?submit=false&form_section=1",
         )
         assert put_response_3.status_code == 200
+
+    def test_audit_columns_are_set_on_create_and_update(self):
+        operator = operator_baker()
+        TestUtils.authorize_current_user_as_operator_user(self, operator)
+        mock_operation = TestUtils.mock_OperationCreateIn()
+        post_response = TestUtils.mock_post_with_auth_role(
+            self, "industry_user", self.content_type, mock_operation.json(), custom_reverse_lazy("create_operation")
+        )
+
+        operation_id = post_response.json().get('id')
+        operation = Operation.objects.get(id=operation_id)
+        assert operation.created_at is not None
+        assert operation.created_by is not None
+        assert operation.updated_at is None
+        assert operation.updated_by is None
+        assert operation.archived_at is None
+        assert operation.archived_by is None
+
+        updated_mock_operation = TestUtils.mock_OperationUpdateIn()
+        updated_mock_operation.statutory_declaration = MOCK_DATA_URL
+
+        TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user',
+            self.content_type,
+            updated_mock_operation.json(),
+            custom_reverse_lazy("update_operation", kwargs={"operation_id": operation.id})
+            + "?submit=false&form_section=1",
+        )
+
+        # check operation audit columns
+        operation.refresh_from_db()  # refresh the operation object to get the updated audit columns
+        assert operation.created_at is not None
+        assert operation.created_by is not None
+        assert operation.updated_at is not None
+        assert operation.updated_by is not None
+        assert operation.archived_at is None
+        assert operation.archived_by is None
+
+        # check document audit columns
```

Comment:
I don't think there is any way to update a document in the app right now. If a user replaces a file, then we get a new document record.

Reply:
Sorry, ignore my comment. I wasn't paying attention to the rest of the code.